internal/provisioner/kops_provisioner.go

```diff
@@ -208,33 +300,53 @@ func (provisioner *KopsProvisioner) CreateCluster(cluster *model.Cluster, aws aw
 		mattermostOperatorNamespace,
 	}
 
-	_, err = k8sClient.CreateNamespaces(namespaces)
+	// Remove all previously-installed operator namespaces and resources.
+	for _, namespace := range namespaces {
+		logger.Infof("Cleaning up namespace %s", namespace)
+		err = k8sClient.Clientset.CoreV1().Namespaces().Delete(namespace, &metav1.DeleteOptions{})
+		if k8sErrors.IsNotFound(err) {
+			logger.Infof("Namespace %s not found; skipping...", namespace)
+		} else if err != nil {
+			return errors.Wrapf(err, "failed to delete namespace %s", namespace)
```

Comment:
Does it make sense to fail to provision if we couldn't delete something?

Reply:
In this case it does. We are trying to remove the namespace to entirely cleanup the old operator resources. If this fails, we will have a problem when we immediately try to recreate a namespace with the same name a few lines down.